components/textdiff/textdiff.js

```diff
@@ -162,28 +161,30 @@ class TextDiffViewModel {
         drawFileList: false,
       });
 
-      this.numberOfSelectedPatchLines = 0;
       let index = 0;
 
       // ko's binding resolution is not recursive, which means below ko.bind refresh method doesn't work for
       // data bind at getPatchCheckBox that is rendered with "html" binding.
       // which is reason why manually updating the html content and refreshing kobinding to have it render...
       if (this.patchLineList) {
-        html = html.replace(/<span class="d2h-code-line-[a-z]+">(\+|-)/g, (match, capture) => {
-          if (this.patchLineList()[index] === undefined) {
-            this.patchLineList()[index] = true;
+        html = html.replace(
+          /<td class="(d2h-(?:del|ins)[^"]*)">([\s\S]*?)<\/td>/g,
+          (match, tdClass, content) => {
+            if (this.patchLineList()[index] === undefined) {
+              this.patchLineList()[index] = true;
+            }
+            content = content.replace(
+              /<span class="d2h-code-line-[a-z]+">(\+|-)<\/span>/g,
+              (match, capture) => this.getPatchCheckBox(capture, index)
+            );
+            return `<td class="${tdClass}" data-bind="click: (editState() === 'patched') ? togglePatchLine.bind($data, ${index++}) : null, css: { patched: editState() === 'patched' }">${content}</td>`;
           }
-
-          return this.getPatchCheckBox(capture, index, this.patchLineList()[index++]);
-        });
+        );
       }
 
-      if (html !== this.htmlSrc) {
```

Comment:
> Unfortunatley this makes selecting lines a bit hard. Because the selection gets lost after the `detectReActivity` mouse move delay...

Yeah, that's an additional problem I hadn't noticed, thanks (sigh).

> Maybe we will need the `checked` attribute on the checkbox again to force the HTML to be different in your described scenario?

This won't work, as demonstrated by the fact that the bug I described in my last post is present in the v1.5.11 release where this attribute is still written.  The `html == this.htmlSrc` check compares the html the last time the render function ran against the newly generated html -- in the scenario above, the `editState` is set to `'none'` when all the boxes are unchecked, and the `patchLineList` array is emptied (not set to all `false`, for example), and so the `TextDiffViewModel.render` method recreates it will all values set to true.  The result is the html string contains all `checked` attributes on all checkboxes, and is identical to the last time the component rendered.

I'll give it some more thought; the whole way the `patchLineList` is connected with the `editState` and instantiated in `textdiff.js` and invalidated in `staging.js` could stand a rethink, and I'm starting to get far more familiar with the architecture here and also knockout.js than I ever intended.

 Fwiw none of these problems, except the bug already in production, exist with my original PR as submitted, and I'll be continuing to use it that way locally, as I have been for months now.